src/advancedKubectl/content.md

```diff
@@ -0,0 +1,1108 @@
+
+
+### Kubernetes internals
+
+Kubernetes consists of a set of independent components that run as separate processes on the nodes of a cluster. Some components run on the master nodes and others run on the worker nodes, and each component has a very specific function.
+
+The most important components on the master nodes are:
+
+- **Storage backend:** stores resource definitions (usually [etcd](https://coreos.com/etcd/) is used)
+- **API server:** provides Kubernetes API and manages storage backend
+- **Controller manager:** ensures resource statuses match specifications
+- **Scheduler:** schedules Pods to worker nodes
+
+And the most important component running on each worker node is:
+
+- **kubelet:** manages execution of containers on a worker node
+
+To see how these components work together, let's consider an example.
+
+Assume, you just executed `kubectl create -f replicaset.yaml`, upon which kubectl made an HTTP POST request to the [*create ReplicaSet* API endpoint](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#create-replicaset-v1-apps) (passing along your ReplicaSet resource definition).
+
+What happens now inside Kubernetes?
+
+
+```slideshow
+{
+  "description": "Walkthrough from the creation of a ReplicaSet to the execution of the containers on the worker nodes.",
+  "slides": [
+    {
+      "image": "kubernetes-internals-1.svg",
+      "description": "Following `kubectl create -f replicaset.yaml`, the API server saves your ReplicaSet resource definition in the storage backend."
+    },
+    {
+      "image": "kubernetes-internals-2.svg",
+      "description": "This triggers the ReplicaSet controller in the controller manager, who watches for creations, updates, and deletions of ReplicaSet resources."
+    },
+    {
+      "image": "kubernetes-internals-3.svg",
+      "description": "The ReplicaSet controller creates a Pod definition for each replica of the ReplicaSet (according to the Pod template in the ReplicaSet definition) and saves them in the storage backend."
+    },
+    {
+      "image": "kubernetes-internals-4.svg",
+      "description": "This triggers the scheduler who watches for Pods that have not yet been assigned to a worker node."
+    },
+    {
+      "image": "kubernetes-internals-5.svg",
+      "description": "The scheduler chooses a suitable worker node for each Pod and adds this information to the Pod definitions in the storage backend."
+    },
+    {
+      "image": "kubernetes-internals-6.svg",
+      "description": "This triggers the kubelet on the worker node that the Pods have been scheduled to, who watches for Pods that have been scheduled to its worker node."
+    },
+    {
+      "image": "kubernetes-internals-7.svg",
+      "description": "The kubelet reads the Pod definitions from the storage backend, downloads the required container images, and runs the containers via Docker (or another container runtime) on the worker node."
```

Comment:
_delegates running the containers to Docker_

Reply:
Thanks, there was actually a mistake, as of course the container runtime downloads the images and not the kubelet.